src/dataflow/analysis/tests/analysis-test.ts

```diff
@@ -42,9 +43,51 @@ function createChainOfEdges(...nodeIds: string[]) {
     const secondNode = nodes[i];
     edges.push(new TestEdge(firstNode, secondNode, firstNode.nodeId + '->' + secondNode.nodeId));
   }
+
+  // Mark the first node in the chain as having ingress, so the whole chain is
+  // valid. 
+  nodes[0].ingress = true;
```

Comment:
Might be worth updating the comment for this function to reflect the ingress point... happy to be updated later.